pycm/pycm_obj.py

```diff
@@ -83,7 +86,10 @@ def __init__(
         __obj_assign_handler__(self, matrix_param)
         __class_stat_init__(self)
         __overall_stat_init__(self)
-        self.imbalance = imbalance_check(self.P)
+        if self.imbalance is None:
+            if is_imbalanced is None:
+                is_imbalanced = imbalance_check(self.P)
+            self.imbalance = is_imbalanced
         self.binary = binary_check(self.classes)
         self.recommended_list = statistic_recommend(self.classes, self.P)
```

Comment:
@sadrasabouri Please take a look at the `statistic_recommend` function, you should update it according to our new system.